File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        /**
            Sorted in ascending order
            for res idx1 < idx2 and we are returing a list with two value
            idx1 != idx2
            start left 
            start right 
            move the right pointer if sum > target 
            move the left pointer if sum < target
        **/

        int ptr1 = 0;
        int ptr2 = numbers.length - 1;

        while(ptr1 < ptr2){
            int sum = numbers[ptr1] + numbers[ptr2];
            if(sum < target){
                ptr1++;
            }else if(sum > target){
                ptr2--;
            }else{
                return new int[]{ptr1+1,ptr2+1};
            }
        }

        return null;
    }
}

File: Data Structures & Algorithms/valid-palindrome-ii/submission-3.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        left = 0
        right = len(s) - 1

        def isPal(l, r):
            while l < r :
                if s[l] != s[r]:
                    return False
                l+=1
                r-=1

            return True


        while left < right:
            if s[left] == s[right]:
                left +=1
                right -= 1
                continue

            if isPal(left+1, right) or isPal(left, right-1):
                return True
            else:
                return False
            


        return True